File: iris-ld/iris-ld-proofs/src/main/java/city/smartb/iris/ldproof/VerifiableJsonLd.java
package city.smartb.iris.ldproof;

import city.smartb.iris.jsonld.JsonLdObject;
import city.smartb.iris.jsonld.reader.JsonFieldReader;
import city.smartb.iris.ldproof.serializer.VerifiableJsonLdDeserializer;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;

import java.util.Map;

@JsonDeserialize(using = VerifiableJsonLdDeserializer.class)
public class VerifiableJsonLd extends JsonLdObject {

    public VerifiableJsonLd(Map<String, Object> jsonLdObject) {
        super(jsonLdObject);
    }

    public VerifiableJsonLd(Map<String, Object> jsonLdObject, JsonFieldReader fieldReader) {
        super(jsonLdObject, fieldReader);
    }

    public LdProof getProof() {
        Map<String, Object> json = get(LdProof.JSON_LD_PROOF).asMap();
        return LdProof.fromMap(json);
    }
}

File: iris-ld/iris-ld-proofs/src/main/kotlin/city/smartb/iris/ld/ldproof/VerifiableJsonLd.kt
package city.smartb.iris.ld.ldproof

import city.smartb.iris.ld.jsonld.JsonLdObject
import city.smartb.iris.ld.jsonld.reader.JsonFieldReader
import city.smartb.iris.ld.ldproof.serializer.VerifiableJsonLdDeserializer
import com.fasterxml.jackson.databind.annotation.JsonDeserialize

@JsonDeserialize(using = VerifiableJsonLdDeserializer::class)
class VerifiableJsonLd : JsonLdObject {
    constructor(jsonLdObject: Map<String, Any>) : super(jsonLdObject)
    constructor(jsonLdObject: Map<String, Any>, fieldReader: JsonFieldReader) : super(jsonLdObject, fieldReader)

    val proof: LdProof
        get() {
            val json: Map<String, Any> = get(LdProof.JSON_LD_PROOF).asMap()
            return LdProof.fromMap(json)
        }
}
